### PR TITLE
[SPARK-40010][PYTHON][DOCS][FOLLOWUP] Make pyspark.sql.window examples self-contained (part 2)

### PR DESCRIPTION
**Pull request by dcoliversun**
### What changes were proposed in this pull request?

As mentioned [here](https://issues.apache.org/jira/browse/SPARK-40148),  we need have examples for several API such as orderBy in `pyspark.sql.window`.

### Why are the changes needed?

To make the documentation more readable and able to copy and paste directly in PySpark shell.

### Does this PR introduce _any_ user-facing change?

Yes, Documentation changes only.

### How was this patch tested?

Github Action about python doctest.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
Looks pretty good. cc @itholic @Yikun in case you guys find some time to review.

**Comment by dcoliversun** [comment]
@HyukjinKwon Hi. There are no examples for four method in `WindowSpec`, e.g. `partitionBy`/`orderBy`/`rowsBetween`/`rangeBetween`.  I believe these are same with methods in `Window`. Do we need to add examples in `WindowSpec` ?

**Comment by HyukjinKwon** [comment]
I think that's fine without it. `WindowSpec` is supposed to be created with `Window` anyway.

**Comment by dcoliversun** [comment]
OK. This PR is ready to review :)

File: python/pyspark/sql/window.py [hunk 1]
```diff
@@ -95,6 +133,45 @@ def orderBy(*cols: Union["ColumnOrName", List["ColumnOrName_"]]) -> "WindowSpec"
         ----------
         cols : str, :class:`Column` or list
             names of columns or expressions
+
```
Comment (HyukjinKwon) [inline python/pyspark/sql/window.py:136]:
```suggestion
```

File: python/pyspark/sql/window.py [hunk 2]
```diff
@@ -371,7 +460,7 @@ def _test() -> None:
     globs["spark"] = spark
 
     (failure_count, test_count) = doctest.testmod(
-        pyspark.sql.window, globs=globs, optionflags=doctest.NORMALIZE_WHITESPACE
+        pyspark.sql.window, globs=globs, optionflags=doctest.ELLIPSIS | doctest.NORMALIZE_WHITESPACE
```
Comment (HyukjinKwon) [inline python/pyspark/sql/window.py:463]:
```suggestion
        pyspark.sql.window, globs=globs, optionflags=doctest.NORMALIZE_WHITESPACE
```

**Review by HyukjinKwon — APPROVED** [review]
LGTM

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by dcoliversun** [comment]
I'm glad for your help @HyukjinKwon @itholic

**Pull request closed without merging**